lib/temporal/activity/poller.rb

```diff
@@ -21,16 +21,21 @@ def initialize(namespace, task_queue, activity_lookup, config, middleware = [],
         @config = config
         @middleware = middleware
         @shutting_down = false
+        @shutting_down_mutex = Mutex.new
```

Comment:
I'm not sure this mutex is actually needed since it only wraps accessing the `@shutting_down` instance variable, which itself is pretty much atomic and won't cause a context switch

Reply:
I'm happy to remove it if you don't think it's necessary